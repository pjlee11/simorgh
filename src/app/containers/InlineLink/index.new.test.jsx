import React from 'react';
import { StaticRouter } from 'react-router-dom';
import {
  shouldMatchSnapshot,
  shouldShallowMatchSnapshot,
} from '../../helpers/tests/testHelpers';
import InlineLinkContainer from './index.new';

const fragmentBlock = (text, attributes = []) => ({
  type: 'fragment',
  model: {
    text,
    attributes,
  },
});

const testInternalInlineLink = (description, locator, blocks) => {
  shouldMatchSnapshot(
    description,
    /*
      for the value it would bring, it is much simpler to wrap a react-router Link in a Router, rather than mock a Router or pass some mocked context.
    */
    <StaticRouter>
      <InlineLinkContainer locator={locator} blocks={blocks} />
    </StaticRouter>,
  );
};

describe('InlineLinkContainer', () => {
  describe('with link matching routes for SPA', () => {
    testInternalInlineLink(
      'should render correctly',
      'https://www.bbc.com/news/articles/c85pqyj5m2ko',
      [fragmentBlock('This is text for an internal link')],
    );

    testInternalInlineLink(
      'should render correctly for TEST environment',
      'https://www.test.bbc.com/news/articles/c85pqyj5m2ko',
      [fragmentBlock('This is text for an internal link')],
    );
  });

  describe('with link not matching SPA route', () => {
    shouldShallowMatchSnapshot(
      'should render correctly',
      <InlineLinkContainer
        locator="https://www.bbc.com/news"
        blocks={[fragmentBlock('This is bold text for a link', ['bold'])]}
      />,
    );
  });
});
