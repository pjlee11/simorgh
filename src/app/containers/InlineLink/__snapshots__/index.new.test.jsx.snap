// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InlineLinkContainer with link matching routes for SPA should render correctly 1`] = `
.c0 {
  color: #0F556C;
  padding: 2px;
  border-bottom: 1px solid #0F556C;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:visited {
  color: #5C5752;
  border-bottom: 1px solid #5C5752;
}

.c0:focus {
  background-color: #0F556C;
  border-bottom: 1px solid #F5F3F1;
  color: #F5F3F1;
}

.c0:hover {
  color: #0F556C;
  background-color: #C3DEE7;
  border-bottom: 1px solid #0F556C;
}

<a
  className="-Link c0"
  href="/news/articles/c85pqyj5m2ko"
  onClick={[Function]}
>
  This is text for an internal link
</a>
`;

exports[`InlineLinkContainer with link matching routes for SPA should render correctly for TEST environment 1`] = `
.c0 {
  color: #0F556C;
  padding: 2px;
  border-bottom: 1px solid #0F556C;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:visited {
  color: #5C5752;
  border-bottom: 1px solid #5C5752;
}

.c0:focus {
  background-color: #0F556C;
  border-bottom: 1px solid #F5F3F1;
  color: #F5F3F1;
}

.c0:hover {
  color: #0F556C;
  background-color: #C3DEE7;
  border-bottom: 1px solid #0F556C;
}

<a
  className="-Link c0"
  href="/news/articles/c85pqyj5m2ko"
  onClick={[Function]}
>
  This is text for an internal link
</a>
`;

exports[`InlineLinkContainer with link not matching SPA route should render correctly 1`] = `
<InlineLink
  href="https://www.bbc.com/news"
>
  <Blocks
    blocks={
      Array [
        Object {
          "model": Object {
            "attributes": Array [
              "bold",
            ],
            "text": "This is bold text for a link",
          },
          "type": "fragment",
        },
      ]
    }
    componentsToRender={
      Object {
        "fragment": [Function],
      }
    }
  />
</InlineLink>
`;
